resources/qml/Settings/SettingTextField.qml

```diff
@@ -98,9 +98,9 @@ SettingItem
 
             selectByMouse: true;
 
-            maximumLength: 10;
+            maximumLength: (definition.type == "[int]") ? 20 : 10;
 
-            validator: RegExpValidator { regExp: (definition.type == "int") ? /^-?[0-9]{0,10}$/ : /^-?[0-9]{0,9}[.,]?[0-9]{0,10}$/ } // definition.type property from parent loader used to disallow fractional number entry
+            validator: RegExpValidator { regExp: (definition.type == "[int]") ? /^\[(\s*-?[0-9]+\s*,)*(\s*-?[0-9]+)\s*\]$/ : (definition.type == "int") ? /^-?[0-9]{0,10}$/ : /^-?[0-9]{0,9}[.,]?[0-9]{0,10}$/ } // definition.type property from parent loader used to disallow fractional number entry
```

Comment:
It seems that the problem is that this only accepts proper final values, not intermediary values while the user is typing. The only reason why the brackets are not a problem is because the default value is an empty lists with pre-filled brackets. If the user would start off without brackets, he could never type anything.

I think the solution is to allow zero-length integers in this field, then interpret that as the number 0 (or erase it from the list). Actually replacing all the `+` in the regex with `{0,10}` would fix it.